APIGOV-17331 - Add instruction for Azure traceability agent (#1563)
Axway/axway-open-docs: content/en/docs/central/connect-azure-gateway/_index.md

### Prepare AMPLIFY Central

Create a **service account** using AMPLIFY Central command: `amplify central create service-account`. This command generates a key pair (public_key.pem / private_key.pem) and associates the public key to an AMPLIFY Central Service account. The service account client ID is required for the agent configuration. Refer to AMPLIFY Central UI / Access / Service Accounts. You will also need the generated keys for starting the agents.

Create an **environment** using the AMPLIFY Central command:  `amplify central create environment my-azure-env`. This command creates an environment object named 'my-azure-env' in AMPLIFY Central. It will become the placeholder for the discovered APIs. This environment name will be part of the agent configuration.

## Connect Azure API Management services to AMPLIFY Central for discovery

The following is a high-level overview of the required steps to connect Azure API Management services to AMPLIFY Central:

* Create an Azure Service principal
* Create a service account for the agent to communicate with the AMPLIFY platform
* Create an environment to group the APIs

> * Pull the discovery agent from Docker

Reviewer: ```suggestion
* Pull the Discovery Agent from Docker
```

* Update the Azure agent YAML file or use an environment variable file to configure the access to Azure.

> ### Creating Azure Service principal

Reviewer: ```suggestion
### Creating an Azure service principal
```

You can retrieve your subscription id with the command: `az account show --query id`

> ### Prepare discovery agent configuration

Reviewer: ```suggestion
### Prepare Discovery Agent configuration
```

First, pull out the image: `docker pull beano.swf-artifactory.lab.phx.axway.int/beano/azure-discovery-agent`.

Second, run the following docker command: `docker run --env-file <PATH>/da_env_vars.env -v <PATH>/<SA-Key-folder>:/keys beano.swf-artifactory.lab.phx.axway.int/beano/azure-discovery-agent`.

Note: `SA-Key-folder` is the folder name that contains the service account keys (private_key.pem / public_key.pem).

## Connect Azure API Management services to AMPLIFY Central for traceability

The following is a high-level overview of the required steps to connect Azure API Management services to AMPLIFY Central:

> * Create an Azure Event Hub Namespace and Event Hub

Reviewer: ```suggestion
* Create an Azure Event Hubs namespace and event hub
```

Reviewer 2: Event Hub is an Azure service and Microsoft spell it with upercase letter for each words.

> * Create an Azure Diagnostic Setting

Reviewer: ```suggestion
* Create an Azure diagnostic setting
```

* Configure Azure API's for Azure monitoring

> * Pull the traceability agent from Docker

Reviewer: ```suggestion
* Pull the Traceability Agent from Docker
```

* Update the Azure agent YAML file or use an environment variable file to configure the access to Azure.

> ### Creating Azure Event Hub Namespace and Event Hub

Reviewer: ```suggestion
### Creating Azure Event Hubs namespace and event hub
```

> Azure Event Hubs is a big data streaming platform and event ingestion service. Refer to [https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-about](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-about)

Reviewer: ```suggestion
Azure Event Hubs is a big data streaming platform and event ingestion service. Refer to [https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-about](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-about).
```

> In order to create an Azure event hub, it is first required to create an event hub namespace. For that, go to Azure Portal and select the \[Event Hubs\] service. Then create the event hub nanmspace. Select the appropriate Azure subscription and resource group where this namespace will be available. Be sure to create the event hub namespace in the same region as your API Management service. For the pricing, we recommend the Standard one. When creating the event hb namespace, Azure will automatically attach to it an access policy. We recommend to keep the default values provided.

Reviewer: ```suggestion
In order to create an Azure event hub, it is first required to create an Event Hubs namespace. For that, go to Azure Portal and select the \[Event Hubs\] service. Then create the Event Hubs namespace. Select the appropriate Azure subscription and resource group where this namespace will be available. Be sure to create the Event Hubs namespace in the same region as your API Management service. For the pricing, we recommend the Standard one. When creating the Event Hubs namespace, Azure will automatically attach to it an access policy. We recommend to keep the default values provided.
```

> Once the namespace is created, you can add an event Hub to this namespace.

Reviewer: ```suggestion
Once the namespace is created, you can add an event hub to this namespace.
```

Reviewer 2: Event Hub instead

> What you need to remember from this event hub creation:

Reviewer: Remember the following when creating the event hub, as you'll need the information for the Traceability Agent configuration:

Reviewer 2: Updated

> * the **event hub namespace name**

Reviewer: ```suggestion
* the **Event Hubs namespace name**
```

* the **event hub name**

> * the **policy name** attached to the event hub namespace (RootManageSharedAccessKey by default)

Reviewer: ```suggestion
* the **policy name** attached to the Event Hubs namespace (RootManageSharedAccessKey by default)
```

* the **policy primary or second key**

> 

Reviewer: Remove "the" from each of the above bullets

Reviewer 2: updated

> Traceability Agent will need these information in its configuration.

Reviewer: Remove this sentence. Reword line #174.

> ### Creating Azure Diagnostic Settings

Reviewer: ```suggestion
### Creating Azure diagnostic settings
```

> The Azure Diagnostic settings will link the Resource group to the Event Hub.

Reviewer: ```suggestion
The Azure diagnostic settings will link the Resource group to the Event Hub.
```

> For creating it, go to Azure portal, open your resource group and open the Monitoring / Diagnostic settings page. Simply Add a diagnostic setting from there. Give it a name and select "GatewayLogs" and "Stream to an event hub" options. The "Stream to event hub option" requires the previous information from event hub namespace and event hub name. Save your changes.

Reviewer: ```suggestion
To create a diagnostic setting: 
1. Go to Azure portal, open your resource group and open the Monitoring / Diagnostic settings page. 
2. Add a diagnostic setting and give it a name. 
3. Select the **GatewayLogs** and **Stream to an event hub** options. The **Stream to event hub** option requires the previous information from Event Hubs namespace and event hub name. 
4. Save your changes.
```

> Now Azure is ready to register your APi traffic to the event hub.

Reviewer: ```suggestion
Azure is now ready to register your API traffic to the event hub.
```

> ### Configuring API Azure Monitoring

Reviewer: ```suggestion
### Configure API Azure monitoring
```

> By default Azure does not monior any API traffic nor their headers. You have to explicitly ask for that. You can do it per API or for all APIs present in the API Managment service.

Reviewer: ```suggestion
By default, Azure does not monitor API traffic or their headers. You have to explicitly ask for that. You can do it per API or for all APIs present in the API Management service.
```

> For this last step, open the settings of an API (or All APIs) and select the Azure Monitor tab. Enable it, select the Vebose mode and add all headers you want to track. These headers can belongs to any request/response (frontend/backend) or you can decide to have different header to log using the advanced options.

Reviewer: ```suggestion
To configure monitoring: 
1. Open the settings of an API (or all APIs). 
2. Select the Azure Monitor tab. 
3. Enable it. 
4. Select the Verbose mode and add all headers you want to track. These headers can belongs to any request/response (frontend/backend) or you can choose different headers to log using the advanced options.
```

Reviewer: I don't understand Step 3, "Enable it."

Reviewer 2: there is a check box to enable the Monitoring. If uncheck, no monitor.

> ### Prepare traceability agent configuration

Reviewer: ```suggestion
### Prepare Traceability Agent configuration
```
